modules/matterhorn-workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -1315,26 +1315,29 @@ public void update(final WorkflowInstance workflowInstance) throws WorkflowExcep
         }
       }
 
-      // Before we persist this, extract the metadata
-      final MediaPackage updatedMediaPackage = workflowInstance.getMediaPackage();
-      populateMediaPackageMetadata(updatedMediaPackage);
-      String seriesId = updatedMediaPackage.getSeries();
-      if (seriesId != null && workflowInstance.getCurrentOperation() != null) {
-        // If the mediapackage contains a series, find the series ACLs and add the security information to the
-        // mediapackage
-        try {
+      try {
+
+        // Before we persist this, extract the metadata
+        final MediaPackage updatedMediaPackage = workflowInstance.getMediaPackage();
+
+        populateMediaPackageMetadata(updatedMediaPackage);
+
+        String seriesId = updatedMediaPackage.getSeries();
+        if (seriesId != null && workflowInstance.getCurrentOperation() != null) {
+          // If the mediapackage contains a series, find the series ACLs and add the security information to the
+          // mediapackage
+
           AccessControlList acl = seriesService.getSeriesAccessControl(seriesId);
-          Option<AccessControlList> activeSeriesAcl = authorizationService.getAcl(updatedMediaPackage,
-                 AclScope.Series);
+          Option<AccessControlList> activeSeriesAcl = authorizationService.getAcl(updatedMediaPackage, AclScope.Series);
           if (activeSeriesAcl.isNone() || !AccessControlUtil.equals(activeSeriesAcl.get(), acl))
             authorizationService.setAcl(updatedMediaPackage, AclScope.Series, acl);
-        } catch (SeriesException e) {
-          throw new WorkflowDatabaseException(e);
-        } catch (NotFoundException e) {
-          logger.warn("Series %s not found, unable to set ACLs", seriesId);
-        } catch (Exception e) {
-          logger.error("Error reading ACL from series {}: {}", seriesId, e);
         }
+      } catch (SeriesException e) {
+        throw new WorkflowDatabaseException(e);
+      } catch (NotFoundException e) {
+        logger.warn("Metadata for this mediapackage could not be updated because it wasn't found", e);
```

Comment:
Would it be Ok to add the mpId to the warn log for this and the one below ~ logger.warn("Metadata for mediapackage {} could not be updated because it wasn't found", updatedMediaPackage.getId(), e);

Reply:
Sounds like a good idea, I'll do that!